internal/mode/static/state/graph/httproute.go

```diff
@@ -423,6 +424,61 @@ func validateFilterHeaderModifierFields(
 	return allErrs
 }
 
+func validateFilterResponseHeaderModifier(
+	validator validation.HTTPFieldsValidator,
+	responseHeaderModifier *v1.HTTPHeaderFilter,
+	filterPath *field.Path,
+) field.ErrorList {
+	if errList := validateFilterHeaderModifier(validator, responseHeaderModifier, filterPath); errList != nil {
+		return errList
+	}
+	var allErrs field.ErrorList
+	disallowedResponseHeaderSet := map[string]struct{}{
+		"server":         {},
+		"date":           {},
+		"x-pad":          {},
+		"content-type":   {},
+		"content-length": {},
+		"connection":     {},
+	}
+	invalidPrefix := "x-accel"
+	for _, h := range responseHeaderModifier.Add {
+		valErr := field.Invalid(filterPath.Child("add"), h, "header name is not allowed")
+		name := strings.ToLower(string(h.Name))
+		if _, exists := disallowedResponseHeaderSet[name]; exists {
+			allErrs = append(allErrs, valErr)
+		} else {
+			if strings.HasPrefix(name, strings.ToLower(invalidPrefix)) {
+				allErrs = append(allErrs, valErr)
+			}
+		}
```

Comment:
```suggestion
		if _, exists := disallowedResponseHeaderSet[name]; exists {
			allErrs = append(allErrs, valErr)
			continue
		} 
	        if strings.HasPrefix(name, strings.ToLower(invalidPrefix)) {
		        allErrs = append(allErrs, valErr)
	        }
```

Possible improvement to remove additional else block. Same with the below.